channel/field.go

```diff
@@ -695,47 +695,62 @@ func (inst fieldInstance) sendExcept(p mpacket.Packet, exception mnet.Client) {
 }
 
 func (inst fieldInstance) getRandomSpawnPortal() (portal, error) {
-	portals := []portal{}
+	var spawnPortals []struct {
```

Comment:
I would feel more comfortable re-ordering the portals during NX parsing instead of allocating a new slice on the heap and growing it each time a portal is needed which is a common operation.